File: android/src/main/java/com/flutter/plugin/alibaichuan/AlibaichuanPlugin.java
package com.flutter.plugin.alibaichuan;

import io.flutter.plugin.common.MethodCall;
import io.flutter.plugin.common.MethodChannel;
import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
import io.flutter.plugin.common.MethodChannel.Result;
import io.flutter.plugin.common.PluginRegistry.Registrar;

/** AlibaichuanPlugin */
public class AlibaichuanPlugin implements MethodCallHandler {

  private static LoginServiceHandler loginServiceHandler;

  /** Plugin registration. */
  public static void registerWith(Registrar registrar) {
    final MethodChannel channel = new MethodChannel(registrar.messenger(), "alibaichuan");
    channel.setMethodCallHandler(new AlibaichuanPlugin());

    loginServiceHandler = new LoginServiceHandler(registrar);
  }

  @Override
  public void onMethodCall(MethodCall call, Result result) {
//    if (call.method.equals("getPlatformVersion")) {
//      result.success("Android " + android.os.Build.VERSION.RELEASE);
//    } else {
//      result.notImplemented();
//    }
    switch (call.method) {
      case "initAliBaiChuan":
        // 暂停
        loginServiceHandler.initAliBaiChuan(call, result);
        break;
      case "login":
        // 开始播放
        loginServiceHandler.login(call, result);
        break;
      case "logout":
        loginServiceHandler.logout(call, result);
        break;
      case "isLogin":
        loginServiceHandler.isLogin(call, result);
        break;
      default:
        result.notImplemented();
        break;
    }

  }
}

File: ios/Classes/AlibaichuanPlugin.m
#import "AlibaichuanPlugin.h"
#import "LoginService/init.h"
@implementation AlibaichuanPlugin

LoginServiceHandler *_LoginServiceHandler;

- (id)init:(NSObject <FlutterPluginRegistrar> *)registrar {
    self = [super init];
    if (self) {
        _LoginServiceHandler = [[LoginServiceHandler alloc] init];
    }
    return self;
}

+ (void)registerWithRegistrar:(NSObject<FlutterPluginRegistrar>*)registrar {
    FlutterMethodChannel* channel = [FlutterMethodChannel
                                     methodChannelWithName:@"alibaichuan"
                                     binaryMessenger:[registrar messenger]];
    AlibaichuanPlugin* instance = [[AlibaichuanPlugin alloc] init:registrar];
    [registrar addMethodCallDelegate:instance channel:channel];
}

- (void)handleMethodCall:(FlutterMethodCall*)call result:(FlutterResult)result {
    //  if ([@"getPlatformVersion" isEqualToString:call.method]) {
    //    result([@"iOS " stringByAppendingString:[[UIDevice currentDevice] systemVersion]]);
    //  } else {
    //    result(FlutterMethodNotImplemented);
    //  }
    
    if ([@"initAliBaiChuan" isEqualToString:call.method]) {
        [_LoginServiceHandler initAliBaiChuan:call result:result];
    } else if ([@"isLogin" isEqualToString:call.method]) {
        [_LoginServiceHandler isLogin:call result:result];
    } else if ([@"login" isEqualToString:call.method]) {
        [_LoginServiceHandler login:call result:result];
    } else if ([@"logout" isEqualToString:call.method]) {
        [_LoginServiceHandler logout:call result:result];
    } else{
        result(FlutterMethodNotImplemented);
    }
}

@end

